### PR TITLE
Feature BlacklistedCruxUsers for payment requests

### PR DESCRIPTION
**Pull request by SanchayPahalwani**

### CODE REVIEW

File: src/core/entities/crux-user.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ export interface ICruxUserInformation {
 export interface ICruxUserData {
     configuration: ICruxUserConfiguration;
```
Comment (quixote911) [inline src/core/entities/crux-user.ts]: can we put blacklistAddresses this inside ICruxUserConfiguration?
What is our policy on which kind of attributes go inside ICruxUserConfiguration?

File: src/core/entities/crux-user.ts [hunk 2]
```diff
@@ -91,6 +94,9 @@ export class CruxUser {
     get privateAddresses() {
         return this.cruxUserPrivateAddresses;
     }
+    get blacklist() {
```
Comment (quixote911) [inline src/core/entities/crux-user.ts]: remove this

File: src/core/entities/crux-user.ts [hunk 3]
```diff
@@ -30,6 +30,7 @@ export interface ICruxUserInformation {
 export interface ICruxUserData {
     configuration: ICruxUserConfiguration;
     privateAddresses: ICruxUserPrivateAddresses;
+    blacklistAddresses: string[];
```
Comment (quixote911) [inline src/core/entities/crux-user.ts]: Why is it called blacklistAddresses? We're not blacklisting addresses, we're blacklisting crux IDs

File: src/core/entities/crux-user.ts (hunk 2 above)
Comment (SanchayPahalwani) [reply src/core/entities/crux-user.ts]: This is being used in server like - `const isValidPublisher = checkPublisherAccess(cruxUserById!.blacklist, cruxId);`

File: src/core/entities/crux-user.ts (hunk 3 above)
Comment (SanchayPahalwani) [reply src/core/entities/crux-user.ts]: changed it to `blacklistedCruxUsers`

File: src/core/entities/crux-user.ts (hunk 2 above)
Comment (quixote911) [reply src/core/entities/crux-user.ts]: change name to something like `blacklistedCruxIDs` then. there could be all kinds of blacklists

Comment (quixote911) [reply src/core/entities/crux-user.ts]: No, actually this wouldnt be necessary. blacklist would be accessible via `.config`.

So remove this getter.

File: src/core/entities/crux-user.ts [hunk 4]
```diff
@@ -62,6 +63,7 @@ export class CruxUser {
     private cruxUserConfig!: ICruxUserConfiguration;
     private cruxUserPrivateAddresses!: ICruxUserPrivateAddresses;
     private cruxDomain!: CruxDomain;
+    private blacklistedCruxIDs!: string[];
```
Comment (quixote911) [inline src/core/entities/crux-user.ts]: You don't need this

**Comment by quixote911** [comment]
Please fix PR title

File: src/core/entities/crux-user.ts (hunk 1 above)
Comment (SanchayPahalwani) [reply src/core/entities/crux-user.ts]: Added BlacklistedCruxUsers to CruxUser's configuration

File: src/core/entities/crux-user.ts (hunk 2 above)
Comment (SanchayPahalwani) [reply src/core/entities/crux-user.ts]: Done

File: src/core/entities/crux-user.ts (hunk 4 above)
Comment (SanchayPahalwani) [reply src/core/entities/crux-user.ts]: Removed it

File: src/application/clients/crux-wallet-client.ts [hunk 5]
```diff
@@ -245,6 +245,29 @@ export class CruxWalletClient {
         return {success, failures};
     }
 
+    @throwCruxClientError
+    public addToBlacklist = async (fullCruxIDs: string[]): Promise<{success: string[], failures: string[]}> => {
```
Comment (quixote911) [inline src/application/clients/crux-wallet-client.ts]: make this `blacklistUsers`

Comment (SanchayPahalwani) [reply src/application/clients/crux-wallet-client.ts]: Done

File: src/core/entities/crux-user.ts [hunk 6]
```diff
@@ -71,6 +72,7 @@ export class CruxUser {
         this.setCruxUserConfig(cruxUserData.configuration);
         this.setPublicKey(publicKey);
         this.setCruxUserPrivateAddresses(cruxUserData.privateAddresses);
+        this.setBlacklistedCruxIDs(cruxUserData.configuration.blacklistedCruxUsers);
```
Comment (quixote911) [inline src/core/entities/crux-user.ts]: won't `this.setCruxUserConfig` already cover this????

**Comment by quixote911** [comment]
Approved, not merging at the moment.

**Pull request merged**